.github/workflows/ci.yml

```diff
@@ -0,0 +1,56 @@
+name: Python package
+
+on: [push]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        python-version: [3.6, 3.7, 3.8]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Upgrade pip
+        run: pip install --upgrade pip
+      - name: Install Poetry
+        run: pip install poetry
+      - name: Cache virtual environment
+        uses: actions/cache@v2
+        with:
+          path: .venv
+          key: ${{ runner.os }}-python-${{ matrix.python-version }}-venv-${{ hashFiles('**/poetry.lock') }}
+      - name: Install dependencies
+        run: poetry install --no-interaction
+      - name: pytest
+        run: poetry run pytest --cov=mimesis
+      - name: mypy
+        run: poetry run mypy mimesis/ tests/
+      - name: pyroma
+        run: poetry run pyroma -d .
+      # - name: codecov
+      #   env:
+      #     CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
+      #   run: poetry run codecov --token ${{ env.CODECOV_TOKEN }}
+
+  deploy:
+    if: ${{ github.ref == 'ref/head/master' }}
+    runs-on: ubuntu-latest
+    needs: [build]
+    steps:
+      - run: |
+          poetry python setup.py minify
+          poetry run check-manifest
+          poetry run python setup.py check --restructuredtext --strict
+      - name: Publish
+        env:
+          PYPI_USERNAME: lk-geimfari
+          PYPI_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
+        run: |
+          poetry build
+          poetry publish --username ${{ env.PYPI_USERNAME }} --password ${{ env.PYPI_PASSWORD }}
```

Comment:
I'm not sure that this is secure enough. Will it be printed?

Reply:
GitHub Actions will redact any secrets from the log file so it should be okay.. BUT I looked up the appropriate environment variables for `poetry` and `codecov` and used them instead.

Reply:
Turns out that `codecov` has an [official Action](https://github.com/marketplace/actions/codecov), I had added the appropriate step.

I am not sure the current release strategy but this integration can be further improved by automatically publishing package version to PyPI  based on a GitHub Tagged Release. Right now if will publish on a `push` event to the `master` branch so this PR should probably be changed to merge to the `dev`. This would need to be discussed.

Reply:
@michael0liver Thank you very much! I'll merge this PR as soon, as I will be near to my laptop.

Reply:
@michael0liver Can you, please, change the release strategy to make it publish only when tags are pushed? I think that this is a better strategy for us.

Reply:
Also, maybe we can use `POETRY_PYPI_TOKEN_PYPI`? To publish new packages on PyPi using API token. What do you think?

Reply:
@lk-geimfari I split the workflow files. Now we have `publish.yml` and `test.yml`.
You still have to create some secrets `PYPI_PASSWORD` and `CODECOV_TOKEN`. You can name these whatever you want but make sure to update them in workflow file.

Reply:
> Also, maybe we can use `POETRY_PYPI_TOKEN_PYPI`? To publish new packages on PyPi using API token. What do you think?

Sounds good, I will update the environment variables.

Reply:
@lk-geimfari Should we pin `POETRY_VERSION` in CI ?

Reply:
No, I think it shouldn't

Reply:
@lk-geimfari Okay what is outstanding to get this merged?